### PR TITLE
Further improve sorting algorithm to put undefined values at end of list

### PR DESCRIPTION
**Pull request by snewcomer**
close #374

### CODE REVIEW

File: tests/integration/helpers/sort-by-test.js [hunk 1]
```diff
@@ -44,7 +44,7 @@ module('Integration | Helper | {{sort-by}}', function(hooks) {
       {{~/each~}}
     `);
 
-    assert.equal(find('*').textContent.trim(), 'aAAabccb', 'sorts multiletter words');
+    assert.equal(find('*').textContent.trim(), 'AaaAbccB', 'sorts multiletter words');
```
Comment (snewcomer) [inline tests/integration/helpers/sort-by-test.js:47]: We aren't splitting the strings up by individual char, so I actually think this is now "more" right.

File: addon/helpers/sort-by.js [hunk 2]
```diff
@@ -7,7 +7,11 @@ function normalizeToBoolean(val) {
   }
 
   if (typeof val === 'number') {
-    return val > 0 ? false : true;
+    if (val > 0) {
+      return false;
+    } else if (val < 0) {
+      return true;
+    }
```
Comment (snewcomer) [inline addon/helpers/sort-by.js:14]: We shouldn't swap if `val === 0`, thus the need for this refactor

**Comment by snewcomer** [comment]
@jrjohnson Can I get your careful eye on this PR?  Thank you!!

File: addon/helpers/sort-by.js (hunk 2 above)
Comment (jrjohnson) [reply addon/helpers/sort-by.js:14]: Ah! Yes! This is what I could NOT seem to get my head around and fix. Yay!

File: addon/helpers/sort-by.js [hunk 3]
```diff
@@ -17,6 +21,15 @@ function sortDesc(key, a, b) {
   const aValue = get(a, key);
   const bValue = get(b, key);
 
+  if (typeof bValue == 'undefined') {
```
Comment (jrjohnson) [inline addon/helpers/sort-by.js:24]: This makes more sense to me than my fix.

File: addon/helpers/sort-by.js [hunk 4]
```diff
@@ -17,6 +21,15 @@ function sortDesc(key, a, b) {
...
+    // keep bValue last
+    return -1;
+  }
+  if (typeof aValue == 'undefined') {
+    // put aValue last
+    return 1;
+  }
+
   if (aValue && bValue && aValue.toLowerCase && bValue.toLowerCase) {
```
Comment (jrjohnson) [inline addon/helpers/sort-by.js:33]: `aValue && bValue && ` can now be removed here as it was just to catch `undefined`. I think everything else can have `toLowerCase` check called on it.

File: addon/helpers/sort-by.js [hunk 5]
```diff
@@ -34,6 +47,15 @@ function sortAsc(key, a, b) {
...
```
Comment (jrjohnson) [inline addon/helpers/sort-by.js:59]: Same,  `aValue && bValue && ` can go away now that undfined check is above.

**Comment by snewcomer** [comment]
Thank you for the awesome review!!  4.2.1 released!!

**Pull request merged**